v0/destinations/af/transform.js

```diff
@@ -26,10 +26,20 @@ function responseBuilderSimple(payload, message, destination) {
   const { androidAppId, appleAppId } = destination.Config;
   let endpoint;
   const os = get(message, "context.os.name");
-  if (os && os.toLowerCase() === "android" && androidAppId) {
-    endpoint = `${ENDPOINT}${androidAppId}`;
-  } else if (os && os.toLowerCase() === "ios" && appleAppId) {
-    endpoint = `${ENDPOINT}id${appleAppId}`;
+  if (os && os.toLowerCase() === "android") {
+    if (androidAppId) {
+      endpoint = `${ENDPOINT}${androidAppId}`;
+    } else {
+      throw new CustomError();
+    }
+  } else if (os && os.toLowerCase() === "ios") {
+    if (appleAppId) {
+      endpoint = `${ENDPOINT}id${appleAppId}`;
+    } else {
+      throw new CustomError();
+    }
+  } else if (destination.Config.appId) {
+    endpoint = `${ENDPOINT}${destination.Config.appId}`;
```

Comment:
No changes are required on the transformer as users are expected to fill the `os` info via user transformations. Please refer to the comments added in the other associated PRs for more information.